frontend/Streamlit_GUI.py

```diff
@@ -0,0 +1,68 @@
+import streamlit
+from PIL import Image
+import os
+import requests
+
+
+streamlit.header("Steel surface defect analyser")
+number = streamlit.slider("Pick one of the 100 sample images")
+
+
+curr_path = os.getcwd()
+data_subdir_path = "/Dataset-trafo"
+data_path = curr_path + data_subdir_path
+
+list_namesg = []
+list_namesd = []
+list_samples = []
+for f in os.listdir(data_path + "/good"):
+    list_namesg.append("/good/" + f)
+for f in os.listdir(data_path + "/defect"):
+    list_namesd.append("/defect/" + f)
+for i in range(50):
+    list_samples.append(list_namesg[i])
+    list_samples.append(list_namesd[i])
+data2 = Image.open(data_path + list_samples[number])
```

Comment:
I would use a contenxt manager here (i.e. a `with` block) so that the file is guaranteed to properly close in any case:
```suggestion
with Image.open(data_path + list_samples[number]) as img:
    data2 = img
```

Reply:
Has been changed, but "img.load()" has to be put before "data2 = img" to load the image in memory.